marshal.go

```diff
@@ -1951,16 +1951,22 @@ func unmarshalTuple(info TypeInfo, data []byte, value interface{}) error {
 		}
 
 		for i, elem := range tuple.Elems {
```

Comment:
The logic is duplicated (from before this patch), should we extract a `unmarshalTupleElements()` func?